From b799969e41e864854c60e93f4d51c4e83de9a4be Mon Sep 17 00:00:00 2001
From: Philippe Gervais <philippe.n.gervais@free.fr>
Date: Tue, 25 Apr 2023 17:07:22 +0200
Subject: [PATCH 1891/1936] Fix compilation of zynaddsubfx-ext-gui on MacOS

Removed the 'ugly workaround' which doesn't seem to be necessary
anymore with recent versions of cmake, and added missing link
directory flags that were missing.
---
 src/CMakeLists.txt    | 13 ++-----------
 src/UI/CMakeLists.txt |  3 ++-
 2 files changed, 4 insertions(+), 12 deletions(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 33628a1e..b606ffc2 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -454,17 +454,8 @@ if(NTK_FOUND)
 endif(NTK_FOUND)
 
 if(FltkGui)
-	#UGLY WORKAROUND
-	find_program (FLTK_CONFIG fltk-config)
-	if (FLTK_CONFIG)
-		execute_process (COMMAND ${FLTK_CONFIG} --use-images --ldflags OUTPUT_VARIABLE FLTK_LDFLAGS)
-		string(STRIP "${FLTK_LDFLAGS}" FLTK_LIBRARIES)
-	endif()
-
-	message(STATUS ${FLTK_LDFLAGS})
-
-
-	set(GUI_LIBRARIES zynaddsubfx_gui ${FLTK_LIBRARIES} ${FLTK_LIBRARIES} ${OPENGL_LIBRARIES})
+	
+	set(GUI_LIBRARIES zynaddsubfx_gui ${FLTK_LIBRARIES} ${OPENGL_LIBRARIES})
 
 	add_definitions(-DFLTK_GUI)
 	message(STATUS "Will build FLTK gui")
diff --git a/src/UI/CMakeLists.txt b/src/UI/CMakeLists.txt
index ba977af8..bf3f9c68 100644
--- a/src/UI/CMakeLists.txt
+++ b/src/UI/CMakeLists.txt
@@ -18,7 +18,8 @@ set(UI_fl_files
 )
 
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
-
+# X11_LIBRARY_DIRS is for libXpm.
+link_directories(${LIBLO_LIBRARY_DIRS} ${X11_LIBRARY_DIRS})
 set_source_files_properties(UI/MasterUI.h PROPERTIES GENERATED 1)
 fltk_wrap_ui(zynaddsubfx_gui ${UI_fl_files})
 
-- 
2.47.0

